language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/TypeParameterContextProvider.java

```diff
@@ -108,8 +108,9 @@ public TypeParameterContextProvider(Class<T> attachmentPoint) {
             if (symbol.kind() != SymbolKind.TYPE_DEFINITION) {
                 return false;
             }
-            TypeDescKind rawType = CommonUtil.getRawType(((TypeDefinitionSymbol) symbol).typeDescriptor()).typeKind();
-            return rawType == TypeDescKind.MAP || rawType == TypeDescKind.RECORD;
+            TypeSymbol rawType = CommonUtil.getRawType(((TypeDefinitionSymbol) symbol).typeDescriptor());
+            return rawType.typeKind() == TypeDescKind.MAP || rawType.typeKind() == TypeDescKind.RECORD ||
+                    CommonUtil.isUnionOfType(rawType, TypeDescKind.RECORD);
```

Comment:
Should we consider union of maps as well?

Reply:
+1